ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+called KeyTable, which is a sharded hash table storing key tokens and reference counts. Access to the KeyTable is threadsafe.
+
+## Specification
+
+### Supported JSON Standard
+
+[RFC7159](http://www.ietf.org/rfc/rfc7159.txt) and [ECMA-404](http://www.ecma-international.org/publications/standards/Ecma-404.htm) 
+JSON data interchange standard is supported. UTF-8 Unicode in JSON text is supported.
+
+### Root JSON Value
+
+In earlier RFC 4627, only objects or arrays were allowed as root values of JSON. Since [RFC 7159](http://www.ietf.org/rfc/rfc7159.txt), 
+the root value of a JSON document can be of any type, scalar type (String, Number, Boolean, Null) or container type (Array, Object). 
+ValkeyJSON is compliant with [RFC 7159](http://www.ietf.org/rfc/rfc7159.txt).
+
+### JSON Command API
+
+There are 22 JSON commands. See https://docs.aws.amazon.com/memorydb/latest/devguide/json-list-commands.html. The API is 
+compatible with RedisJSON v2. Note that API compatibility here means our command API is a superset of RedisJSON API. 
+For example, we have command “JSON.DEBUG DEPTH” and “JSON.DEBUG FIELDS”, while they do not.
+
+### JSONPath Query Syntax
+
+Two kinds of JSONPath query syntaxes are supported:
+* Enhanced syntax – Follows the [Goessner-style](https://goessner.net/articles/JsonPath/) JSONPath query syntax, as shown in the table below.
+* Restricted syntax – Has limited query capabilities, compatible with RedisJSON v1.
+
+If a query path starts with '$', the enhanced syntax is used. Otherwise, the restricted syntax is used. A query using 
+the enhanced syntax always returns an array of values, while a restricted-syntax query always returns a single value. 
+
+![](./images/jsonpath_syntax.png)
+
+For details and examples, see https://docs.aws.amazon.com/memorydb/latest/devguide/json-document-overview.html#json-path-syntax.
+
+### ACL
+
+ValkeyJSON introduces a new ACL category - @json. The category has 22 JSON commands. No existing Valkey commands 
+are members of the @json category. 
+
+There are 4 existing ACL categories which are updated to include new JSON commands: @read, @write, @fast, @slow. The 
+table below has a column for each of these categories and a row for each command. If the cell contains a “y” then that 
+command must be added into that category. All of the other command members of those categories remain unchanged.
+
+![](./images/json_command_acl.png)
+
+### Info metrics
+
+Info metrics are visible through the “info json” or “info modules” command.
+
+| Name                    | 	Description                                                           |
+|:------------------------|:---------------------------------------------------------------------|
+| json_total_memory_bytes | 	Total amount of memory allocated to JSON documents and meta data. |
+| json_num_documents	  | Number of JSON keys                                                   | 
```

Comment:
```suggestion
| json_num_documents	  | Number of JSON keys.                                                  | 
```

Punctuation style should be consistent.  This applies to other tables as well.  Instead of the suggested change, just stop using periods in all these tables.

Reply:
updated